lib/activerecord-tenant-level-security/schema_dumper.rb

```diff
@@ -39,7 +39,7 @@ def policies_in_database
     private
 
     def convert_qual_to_partition_key(qual)
-      matched = qual.match(/^\((.+?) = /)
+      matched = qual.match(/^\(*([a-zA-Z0-9_]+)(?:\)::[a-zA-Z0-9_]+)*\s*=/)
```

Comment:
I'm not sure whether we can assume `a-zA-Z0-9_` as a column name. According to the manual, it seems that more characters can be used 🤔 
https://www.postgresql.org/docs/17/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS

Reply:
It is true that column names may include characters other than the strings specified here. In such cases, column names using special characters are enclosed in double quotes (`""`), and with the current regular expression, column names are not correctly extracted, resulting in an error.

Example:
```
USING (("tenant_id&" = (current_setting('tenant_level_security.tenant_id'::text))::integer))
```

How far we want to support such cases as tenant keys may depend on the policy, but the relevant page states:
```
SQL identifiers and key words must begin with a letter (a-z, but also letters with diacritical marks and non-Latin letters) or an underscore (_). Subsequent characters in an identifier or key word can be letters, underscores, digits (0-9), or dollar signs ($). Note that dollar signs are not allowed in identifiers according to the letter of the SQL standard, so their use might render applications less portable.
```
Given this, I think there is some merit in the idea of limiting the scope to characters compliant with the SQL standard. Ultimately, I believe it would be best to follow the support policy of this library.

Reply:
Fair enough. I believe `a-zA-Z0-9_` is sufficient in many cases, so I am in favor of accepting it.